### PR TITLE
[LIB-688] Added methods to handle Data Point fetch

### PR DESCRIPTION
**Pull request by jaskooo**

### CODE REVIEW

File: syncano-ios/SCAPIClient+SCDataObject.h [hunk 1]
```diff
@@ -37,6 +37,15 @@ NS_ASSUME_NONNULL_BEGIN
...
+- (void)getDataObjectsFromViewName:(NSString *)viewName params:(nullable NSDictionary *)params completion:(SCAPICompletionBlock)completion DEPRECATED_MSG_ATTRIBUTE("Use getDataObjectsFromDataPointWithName:params:completion: method instead");
+
+/**
+ *  Attempts to fetch data object from data point
+ *
+ *  @param viewName   name of a data point
+ *  @param params     parameters
+ *  @param completion completion block
+ */
+- (void)getDataObjectsFromDataPointWithName:(NSString *)dataPointName params:(nullable NSDictionary *)params completion:(SCAPICompletionBlock)completion;
```
Comment (lifcio) [inline syncano-ios/SCAPIClient+SCDataObject.h]: DataEndpoint, not DataPoint :)

File: syncano-ios/SCDataObject.h [hunk 2]
```diff
@@ -51,7 +51,16 @@ NS_ASSUME_NONNULL_BEGIN
...
++ (nullable NSString *)viewNameForAPI DEPRECATED_MSG_ATTRIBUTE("Use dataPointNameForAPI method instead");
+
+/**
+ *  Returns data point name used in Syncano API, by default this method returns nil.
+ *  Use it when you want to create a class which is always fetched using a data point.
+ *  When you set dataPointNameForAPI, [YourClass please] will target queries to the view instead of raw class.
+ *
+ *  @return String with API class name
+ */
++ (nullable NSString *)dataPointNameForAPI;
```
Comment (lifcio) [inline syncano-ios/SCDataObject.h]: dataEndpoint

File: syncano-ios/SCDataObject.h [hunk 3]
```diff
@@ -93,7 +102,26 @@ NS_ASSUME_NONNULL_BEGIN
...
++ (SCPlease *)pleaseForView:(NSString*)viewName fromSyncano:(Syncano *)syncano DEPRECATED_MSG_ATTRIBUTE("Use pleaseForDataPoint:fromSyncano: method instead");
+
+/**
+ *  Returns SCPlease instance for Syncano singleton
+ *
+ *  @param dataPointName Name of the Data Point
+ *
+ *  @return SCPlease instance
+ */
++ (SCPlease *)pleaseForDataPoint:(NSString*)dataPointName;
```
Comment (lifcio) [inline syncano-ios/SCDataObject.h]: DataEndpoint

File: syncano-ios/SCDataObject.h [hunk 4]
```diff
@@ -93,7 +102,26 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+/**
+ *  Returns SCPlease instance for provided Syncano instance
+ *
+ *  @param dataPointName Name of the Data Point
+ *  @param syncano Syncano instance, which SCPlease will use to query objects from
+ *
+ *  @return SCPlease instance
+ */
++ (SCPlease *)pleaseForDataPoint:(NSString *)dataPointName fromSyncano:(Syncano *)syncano;
```
Comment (lifcio) [inline syncano-ios/SCDataObject.h]: DataEndpoint

File: syncano-ios/SCPleaseForDataPoint.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+//  Created by Jan Lipmann on 27.06.2016.
+//  Copyright © 2016 Syncano. All rights reserved.
+//
+//
+
+#import "SCPlease.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SCPleaseForDataPoint: SCPlease
```
Comment (lifcio) [inline syncano-ios/SCPleaseForDataPoint.h]: DataEndpoint

**Comment by lifcio** [comment]
Looks good, but change *point to *endpoint, there's more than what I noted.

Another thing - data endpoint works only in API v1.1. data view only in API v1.0. So we need some policy here, how to proceed. You can either add new SCAPIClient, with v1.1 version (like I did in SCScriptEndpoint), or switch default Syncano to v1.1 (but then deprecated SCDataView and SCWebhook will need explicit copy of SCAPIClient with API v1.0)

**Comment by jaskooo** [comment]
@lifcio let's talk about this version problem on our tomorrow sprint meeting. Ther is no easy way to achieve the best solution here.

**Pull request merged**